### ISSUE TITLE
[2019.12] PHP Warnings

### ISSUE THREAD

**Issue by nupplaphil** [friendica/friendica#7676]
Please report here any PHP Warning message obtained running the `2019.09` stable or `2019.12-dev` versions.
Follow-up #7298 

Open Tasks:
 * [ ] https://github.com/friendica/friendica/issues/7676#issuecomment-538666825 `A non-numeric value encountered in .../mod/profile_photo.php on line 72`
 * [ ] https://github.com/friendica/friendica/issues/7676#issuecomment-543261920 `Friendica\Model\GServer::analyseRootBody(): unterminated entity reference            Poke in .../src/Model/GServer.php on line 1064`

**Comment by AlfredSK** [comment]
2019.09 master:
```
PHP Warning:  A non-numeric value encountered in .../mod/profile_photo.php on line 72
PHP Warning:  A non-numeric value encountered in .../mod/profile_photo.php on line 73
PHP Warning:  Imagick::cropimage() expects parameter 3 to be integer, string given in .../src/Object/Image.php on line 612
```

**Comment by AlfredSK** [comment]
2019.12-dev:

```
[17-Oct-2019 14:02:49 Europe/Berlin] PHP Warning:  Friendica\Model\GServer::analyseRootBody(): unterminated entity reference            Poke in .../src/Model/GServer.php on line 1064
[17-Oct-2019 14:02:49 Europe/Berlin] PHP Warning:  Friendica\Model\GServer::analyseRootBody(): unterminated entity reference            Poke in .../src/Model/GServer.php on line 1120
```

**Comment by AndyHee** [comment]
2019.12-dev:

```
PHP Warning:  Friendica\Model\GServer::analyseRootBody(): unterminated entity reference  co-host of the Audacious Compassion podcast. in /var/www/friendica/src/Model/GServer.php on line 1064
PHP Warning:  Friendica\Model\GServer::analyseRootBody(): unterminated entity reference  co-host of the Audacious Compassion podcast. in /var/www/friendica/src/Model/GServer.php on line 1064
PHP Warning:  Friendica\Model\GServer::analyseRootBody(): unterminated entity reference  co-host of the Audacious Compassion podcast. in /var/www/friendica/src/Model/GServer.php on line 1120
```

**Comment by copiis** [comment]
2019.09 master

`PHP Warning:  Error while sending STMT_CLOSE packet. PID=9249 in /src/Database/Database.php on line 652`

**Comment by AndyHee** [comment]
**2019.12-dev**

```
PHP Warning:  Friendica\Model\GServer::analyseRootBody(): unterminated entity reference           Codes in /var/www/friendica/src/Model/GServer.php on line 1120
PHP Warning:  Friendica\Model\GServer::analyseRootBody(): unterminated entity reference  food.
Working at https://machimachi.com
@knu on Twitter & GitHub in /var/www/friendica/src/Model/GServer.php on line 1064
PHP Warning:  Friendica\Model\GServer::analyseRootBody(): unterminated entity reference  food.
Working at https://machimachi.com
@knu on Twitter & GitHub in /var/www/friendica/src/Model/GServer.php on line 1064
PHP Warning:  Friendica\Model\GServer::analyseRootBody(): unterminated entity reference  food.
Working at https://machimachi.com
@knu on Twitter & GitHub in /var/www/friendica/src/Model/GServer.php on line 1120
```

**Comment by AlfredSK** [comment]
2019.12-dev:

`PHP Warning:  is_file() expects parameter 1 to be a valid path, resource given in .../src/Module/Admin/Summary.php on line 88`

**Comment by AlfredSK** [comment]
2019.12-dev:

`PHP Warning:  Use of undefined constant max_csv_file_size - assumed 'max_csv_file_size' (this will throw an Error in a future version of PHP) in .../mod/settings.php on line 1256`

**Comment by annando** [comment]
@AndyHee the warning `MySQL server has gone away` is an indicator for some issue with your MySQL server, but nothing we can fix in Friendica - and we shouldn't suppress this warning, since it indicates something where the admin has to have a look at.

**Comment by AndyHee** [comment]
Thanks @annando noted!

**Comment by AlfredSK** [comment]
2019.12-dev:

```
[08-Nov-2019 22:02:59 UTC] PHP Warning:  call_user_func() expects parameter 1 to be a valid callback, class 'Friendica\Module\Pinned' not found in .../src/App/Module.php on line 252
[08-Nov-2019 22:02:59 UTC] PHP Warning:  call_user_func() expects parameter 1 to be a valid callback, class 'Friendica\Module\Pinned' not found in .../src/App/Module.php on line 256
[08-Nov-2019 22:02:59 UTC] PHP Warning:  call_user_func() expects parameter 1 to be a valid callback, class 'Friendica\Module\Pinned' not found in .../src/App/Module.php on line 264
[08-Nov-2019 22:02:59 UTC] PHP Warning:  call_user_func() expects parameter 1 to be a valid callback, class 'Friendica\Module\Pinned' not found in .../src/App/Page.php on line 311
```

**Comment by AlfredSK** [comment]
2019.12-dev:

`count(): Parameter must be an array or an object that implements Countable in .../src/Module/Profile/Contacts.php on line 49`

**Comment by AlfredSK** [comment]
With the new ACL system I'm getting a lot...

```
[29-Nov-2019 06:53:35 UTC] PHP Warning:  array_unique() expects parameter 1 to be array, null given in /var/www/html/libranet/src/App/Page.php on line 286
```

**Comment by MrPetovan** [comment]
On it.

**Comment by AlfredSK** [comment]
I'm optimistic we will manage this ACL thing until the release. :-)

2019.12-dev:

```
[02-Dec-2019 17:32:56 UTC] PHP Warning:  count(): Parameter must be an array or an object that implements Countable in .../src/Core/ACL.php on line 345
[02-Dec-2019 17:32:56 UTC] PHP Warning:  count(): Parameter must be an array or an object that implements Countable in .../src/Core/ACL.php on line 346
[02-Dec-2019 17:32:56 UTC] PHP Warning:  count(): Parameter must be an array or an object that implements Countable in .../src/Core/ACL.php on line 347
[02-Dec-2019 17:32:56 UTC] PHP Warning:  count(): Parameter must be an array or an object that implements Countable in .../src/Core/ACL.php on line 348
[02-Dec-2019 17:32:56 UTC] PHP Warning:  implode(): Invalid arguments passed in .../src/Core/ACL.php on line 407
[02-Dec-2019 17:32:56 UTC] PHP Warning:  implode(): Invalid arguments passed in .../src/Core/ACL.php on line 408
[02-Dec-2019 17:32:56 UTC] PHP Warning:  implode(): Invalid arguments passed in .../src/Core/ACL.php on line 409
[02-Dec-2019 17:32:56 UTC] PHP Warning:  implode(): Invalid arguments passed in .../src/Core/ACL.php on line 410
```

After the latest fix for ACL and js_upload I pulled from develop and tested an image upload. It works. Image has the correct permissions. But I got those warnings in my PHP log.

**Comment by MrPetovan** [comment]
> I'm optimistic we will manage this ACL thing until the release. :-)

Didn't you mean "before" instead of "until"?

**Comment by AlfredSK** [comment]
Possibly. But deepl.com said 'until release'. Maybe because "bis zur" (until) and "vor der" (before) has the same meaning in German.

**Comment by AlfredSK** [comment]
> I'd like a full trace for the first four messages

Unfortunately, that's all I have. Do you think I can find out more in Friendica's log?

**Comment by MrPetovan** [comment]
Unlikely, this is a language error, not an application one.

**Comment by AlfredSK** [comment]
Looks like #7897 completely fixed https://github.com/friendica/friendica/issues/7676#issuecomment-560503744 - I tried it again w/o any notice/warning/error in PHP log.

**Comment by MrPetovan** [comment]
Yay!

**Comment by copiis** [comment]
```
PHP Warning:  require /vendor/composer/../smarty/smarty/libs/bootstrap.php): failed to open stream: No such file or directory in /vendor/composer/autoload_real.php on line 70
PHP Warning:  require /vendor/composer/../smarty/smarty/libs/bootstrap.php): failed to open stream: No such file or directory in /vendor/composer/autoload_real.php on line 70
PHP Fatal error:  require(): Failed opening required /vendor/composer/../smarty/smarty/libs/bootstrap.php' /vendor/pear/text_languagedetect:.:/usr/share/php') in /vendor/composer/autoload_real.php on line 70

```

**Comment by MrPetovan** [comment]
Please run `bin/composer.phar install --no-dev`

**Comment by copiis** [comment]
`PHP Warning:  Use of undefined constant contact - assumed 'contact' (this will throw an Error in a future version of PHP) in /src/Worker/Delivery.php on line 255`

**Comment by copiis** [comment]
> Please run `bin/composer.phar install --no-dev`

i run that after each git pull

**Comment by MrPetovan** [comment]
@copiis How often this error appears?

**Comment by copiis** [comment]
> @copiis How often this error appears?

once

**Comment by MrPetovan** [comment]
Closed by #7999
